src/razor/src/blazorDebug/blazorDebugConfigurationProvider.ts

```diff
@@ -179,6 +262,220 @@ export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurati
+                }
+                if (proxyUrlString) {
+                    BlazorDebugConfigurationProvider.vsWebAssemblyBridgeOutputChannel.appendLine(
+                        `Debugging proxy is running at: ${proxyUrlString}`
+                    );
+                    const oldPid = BlazorDebugConfigurationProvider.pidsByUrl.get(urlStr);
+                    if (oldPid != undefined) {
+                        process.kill(oldPid);
+                    }
+                    BlazorDebugConfigurationProvider.pidsByUrl.set(urlStr, spawnedProxy.pid);
+                    const url = new URL(proxyUrlString);
+                    return [
+                        `${url.protocol.replace(`http`, `ws`)}//${url.host}{browserInspectUriPath}`,
+                        proxyICorDebugPort,
+                        proxyBrowserPort,
+                    ];
+                }
+            }
+
+            for await (const error of spawnedProxy.stderr) {
+                BlazorDebugConfigurationProvider.vsWebAssemblyBridgeOutputChannel.appendLine(`ERROR: ${error}`);
+            }
+        } catch (error: any) {
+            if (spawnedProxy.pid) {
+                BlazorDebugConfigurationProvider.vsWebAssemblyBridgeOutputChannel.appendLine(
+                    `Error occured while spawning debug proxy. Terminating debug proxy server.`
+                );
+                process.kill(spawnedProxy.pid);
+            }
+            throw error;
+        }
+        return ['', -1, -1];
+    }
+
+    private static async isNet9OrNewer(): Promise<boolean> {
+        let isNet9OrNewer = false;
+        const configurationsObject = {
+            $properties: ['BuildProperties'],
+            BuildProperties: {
+                $properties: ['Id', 'Name', 'Value', 'StorageType'],
+                $where: {
+                    '&&': [
+                        { '==': [{ '.': 'Name' }, 'TargetFramework'] },
+                        { '==': [{ '.': 'StorageType' }, 'ProjectFile'] },
+                    ],
+                },
+            },
+        };
+        const query = {
+            context: 'Projects',
+            query: {
+                $properties: ['ActiveConfigurations'],
+                ActiveConfigurations: configurationsObject,
+                $filter: { ProjectsByCapabilities: ['WebAssembly'] },
+            },
+        };
+
+        const queryString = JSON.stringify(query);
+        const proxy = await getCSharpDevKit()?.exports.serviceBroker.getProxy<IQueryExecutionService>(
+            Descriptors.projectQueryExecutionService
+        );
+        try {
+            if (proxy) {
+                const result = await proxy.ExecuteQueryAsync(queryString);
+                const queryResult = JSON.parse(result) as QueryResult<Project>;
+                if (queryResult && queryResult.Items) {
+                    isNet9OrNewer = true;
+                    queryResult.Items.forEach((project) => {
```

Comment:
Can't we check the project we are trying to launch instead of every project?

Reply:
Ideally yes: At least in general, I don't think it is an uncommon pattern for folks to only bother upgrading only projects in active development, and some folks like to make rather large solutions (and probably workspaces in VS Code, though I have less data there). So, I wouldn't be surprised to see a workspace that has lots of different projects with different versions of .NET. The tooling also encourages that pattern since one needs to start hand-modifying project files to come up with good options to keep all projects on the same target framework.

Reply:
Okay! When we are launching an app without having the assets generated, using the C# devkit I really have the project path that is being launched and I can only look at this project. But when we don't have C# devkit or we have the launch.json generated I don't have this information and then I will loop in all the webassembly projects, does it make sense for you?